### PR TITLE
refactor(autoware_pointcloud_preprocessor): rework random downsample filter parameters

### PR DESCRIPTION
**Pull request by vividf**
## Description
This PR includes the following changes
1. Remove the default parameters for the random downsample filter node. 
2. Add parameter, launch file, and parameters schema for the node. 
3. Change the name `nodelet` to `node`.

A big thank you to @Ariiees for the PRs (https://github.com/autowarefoundation/autoware.universe/pull/8297 and https://github.com/autowarefoundation/autoware.universe/pull/7422), where we worked together on the entire pointcloud preprocessor parameters, launch file, and schema.

## Related links

**Parent Issue:**

- Link

## How was this PR tested?

`ros2 launch autoware_pointcloud_preprocessor random_downsample_filter_node.launch.xml`

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: sensing/autoware_pointcloud_preprocessor/src/downsample_filter/random_downsample_filter_node.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ RandomDownsampleFilterComponent::RandomDownsampleFilterComponent(
 {
   // set initial parameters
   {
-    sample_num_ = static_cast<size_t>(declare_parameter("sample_num", 1500));
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/src/downsample_filter/random_downsample_filter_node.cpp:61]: types do no coincide

**Review by knzo25 — CHANGES_REQUESTED** [review]
Just one data type-related comment

File: sensing/autoware_pointcloud_preprocessor/src/downsample_filter/random_downsample_filter_node.cpp (hunk 1 above)
Comment (vividf) [reply sensing/autoware_pointcloud_preprocessor/src/downsample_filter/random_downsample_filter_node.cpp:61]: @knzo25 thanks! However, declare_parameter<size_t> doesn't work for ros2 parameter. (compile will fatiled)
image

reference: https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/parameters/

Comment (knzo25) [reply sensing/autoware_pointcloud_preprocessor/src/downsample_filter/random_downsample_filter_node.cpp:61]: Same reply as other PRs

Comment (vividf) [reply sensing/autoware_pointcloud_preprocessor/src/downsample_filter/random_downsample_filter_node.cpp:61]: thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/8485/commits/9d3fb7ff15c1fbd004c546b3f6db3e361b99a2dc

**Comment by vividf** [comment]
PRs that need an additional cast for variables: 
1. https://github.com/autowarefoundation/autoware.universe/pull/8487
2. https://github.com/autowarefoundation/autoware.universe/pull/8485
3. https://github.com/autowarefoundation/autoware.universe/pull/8474
4. https://github.com/autowarefoundation/autoware.universe/pull/8469
5. https://github.com/autowarefoundation/autoware.universe/pull/8468

**Comment by vividf** [comment]
@knzo25 kindly ping

**Review by knzo25 — APPROVED** [review]
LGTM

**Pull request merged**